src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -176,61 +207,214 @@ def __blob_is_not_complete(metadata):
+
+    if source_location is None:
+        source_location = "."
+
+    if os.path.exists(source_location):
+        if os.path.isdir(source_location):
+            source_location = _upload_source_code(
+                client, registry_name, resource_group_name, source_location)
+        else:
+            raise CLIError(
+                "'--source-location' should be a local directory path or remote url.")
+    else:
+        source_location = _check_remote_source_code(source_location)
 
-    # [Server Bug] we should allow skiping image_name if users don't want to push
     is_push_enabled = True
     if image_name is None:
         is_push_enabled = False
+        print("'--tag' is not providied. Skip image push after build.")
+    else:
+        image_name = _check_image_name(image_name)
 
     # hard-code platform to linux and cpu to 1
     platform = PlatformProperties("Linux")
     platform.cpu = 1
 
-    # figure out how to pass build_arguments
     build_arguments = []
-    if not (arguments is None):
-        for name_value in arguments:
+    if not (build_args is None):
+        for name_value in build_args:
             name, value = name_value.split('=', 1)
             build_arguments.append(BuildArgument(name, value, False))
 
-    if not (secret_arguments is None):
-        for name_value in secret_arguments:
+    if not (secret_build_args is None):
+        for name_value in secret_build_args:
             name, value = name_value.split('=', 1)
             build_arguments.append(BuildArgument(name, value, True))
 
-    build_request = QueueBuildRequest(
-        image_name=image_name,
-        source_location=source_location,
-        build_parameters=build_parameters,
-        is_push_enabled=is_push_enabled,
-        timeout=timeout,
-        platform=platform,
-        build_arguments=build_arguments)
-
-    result = LongRunningOperation(cmd.cli_ctx)(client.queue(
-        build_request=build_request, resource_group_name=resource_group_name, registry_name=registry_name))
-
-    if isinstance(result, Build):
-        print("Successfully queued a build with build-id: {}. Starting to stream the logs...".format(result.build_id))
-        return acr_build_show_logs(cmd, client, registry_name, result.build_id, resource_group_name)
+    try:
+        build_request = QueueBuildRequest(
+            image_name=image_name,
+            source_location=source_location,
+            build_parameters=build_parameters,
+            is_push_enabled=is_push_enabled,
+            timeout=timeout,
+            platform=platform,
+            build_arguments=build_arguments)
+
+        result = LongRunningOperation(cmd.cli_ctx)(client.queue(
+            build_request=build_request, resource_group_name=resource_group_name, registry_name=registry_name))
+
+        print("Successfully queued a build with build-id: {}.".format(result.build_id))
+
+        if show_logs == 'true':
+            print("Starting to stream the logs...")
```

Comment:
>print("Starting to stream the logs...") [](start = 12, length = 39)

We should reword this to indicate something like  "Build {id} initiated." Log streaming is expected here and so we don't have to explicitly call it out. 
Maybe run a timer that outputs waiting for agent if we don't see the logs in say 3-5 seconds.

Reply:
You are right the output does should the build id before streaming the logs - 
```
uccessfully queued a build with build-id: f32f514dfe1e493080457384a05d7f1d.
Starting to stream the logs...
time="2018-02-15T01:48:36Z" level=info msg="Running command docker login -u 00000000-0000-0000-0000-000000000000 -p *********
```

Reply:
Lets just remove the word "Successfully" - Basically keep it as terse as possible. 
Also we need to indicate that you are zipping stuff up and how much we are uploading ..

Reply:
Updated.